### PR TITLE
Fix wsign conversion

### PR DESCRIPTION
**Pull request by fhuberts**
On top of #835 

See #784

### CODE REVIEW

**Comment by fhuberts** [comment]
Errors in the windows build.
I did not touch that file.

```
D:\a\1\s\src\oatpp\web\server\handler\ErrorHandler.cpp(35,9): warning C4068: unknown pragma 'GCC' [D:\a\1\s\build\src\oatpp.vcxproj]
D:\a\1\s\src\oatpp\web\server\handler\ErrorHandler.cpp(36,9): warning C4068: unknown pragma 'GCC' [D:\a\1\s\build\src\oatpp.vcxproj]
D:\a\1\s\src\oatpp\web\server\handler\ErrorHandler.cpp(49,9): warning C4068: unknown pragma 'GCC' [D:\a\1\s\build\src\oatpp.vcxproj]
D:\a\1\s\src\oatpp\web\server\handler\ErrorHandler.cpp(43,16): warning C4996: 'oatpp::web::server::handler::ErrorHandler::handleError': was declared deprecated [D:\a\1\s\build\src\oatpp.vcxproj]
D:\a\1\s\src\oatpp\web\server\handler\ErrorHandler.cpp(45,16): warning C4996: 'oatpp::web::server::handler::ErrorHandler::handleError': was declared deprecated [D:\a\1\s\build\src\oatpp.vcxproj]
D:\a\1\s\src\oatpp\web\server\handler\ErrorHandler.cpp(47,16): warning C4996: 'oatpp::web::server::handler::ErrorHandler::handleError': was declared deprecated [D:\a\1\s\build\src\oatpp.vcxproj]
D:\a\1\s\src\oatpp\web\server\handler\ErrorHandler.cpp(57,9): warning C4068: unknown pragma 'GCC' [D:\a\1\s\build\src\oatpp.vcxproj]
D:\a\1\s\src\oatpp\web\server\handler\ErrorHandler.cpp(58,9): warning C4068: unknown pragma 'GCC' [D:\a\1\s\build\src\oatpp.vcxproj]
D:\a\1\s\src\oatpp\web\server\handler\ErrorHandler.cpp(60,9): warning C4068: unknown pragma 'GCC' [D:\a\1\s\build\src\oatpp.vcxproj]
D:\a\1\s\src\oatpp\web\server\handler\ErrorHandler.cpp(59,10): warning C4996: 'oatpp::web::server::handler::ErrorHandler::handleError': was declared deprecated [D:\a\1\s\build\src\oatpp.vcxproj]
```

Will submit a separate PR to hide those GCC pragmas behind '#ifdef __GNUC__'

**Comment by fhuberts** [comment]
rebased on #835, which was rebased on #837

**Comment by fhuberts** [comment]
> Errors in the windows build. I did not touch that file.
> 
> ```
> D:\a\1\s\src\oatpp\web\server\handler\ErrorHandler.cpp(35,9): warning C4068: unknown pragma 'GCC' [D:\a\1\s\build\src\oatpp.vcxproj]
> D:\a\1\s\src\oatpp\web\server\handler\ErrorHandler.cpp(36,9): warning C4068: unknown pragma 'GCC' [D:\a\1\s\build\src\oatpp.vcxproj]
> D:\a\1\s\src\oatpp\web\server\handler\ErrorHandler.cpp(49,9): warning C4068: unknown pragma 'GCC' [D:\a\1\s\build\src\oatpp.vcxproj]
> D:\a\1\s\src\oatpp\web\server\handler\ErrorHandler.cpp(43,16): warning C4996: 'oatpp::web::server::handler::ErrorHandler::handleError': was declared deprecated [D:\a\1\s\build\src\oatpp.vcxproj]
> D:\a\1\s\src\oatpp\web\server\handler\ErrorHandler.cpp(45,16): warning C4996: 'oatpp::web::server::handler::ErrorHandler::handleError': was declared deprecated [D:\a\1\s\build\src\oatpp.vcxproj]
> D:\a\1\s\src\oatpp\web\server\handler\ErrorHandler.cpp(47,16): warning C4996: 'oatpp::web::server::handler::ErrorHandler::handleError': was declared deprecated [D:\a\1\s\build\src\oatpp.vcxproj]
> D:\a\1\s\src\oatpp\web\server\handler\ErrorHandler.cpp(57,9): warning C4068: unknown pragma 'GCC' [D:\a\1\s\build\src\oatpp.vcxproj]
> D:\a\1\s\src\oatpp\web\server\handler\ErrorHandler.cpp(58,9): warning C4068: unknown pragma 'GCC' [D:\a\1\s\build\src\oatpp.vcxproj]
> D:\a\1\s\src\oatpp\web\server\handler\ErrorHandler.cpp(60,9): warning C4068: unknown pragma 'GCC' [D:\a\1\s\build\src\oatpp.vcxproj]
> D:\a\1\s\src\oatpp\web\server\handler\ErrorHandler.cpp(59,10): warning C4996: 'oatpp::web::server::handler::ErrorHandler::handleError': was declared deprecated [D:\a\1\s\build\src\oatpp.vcxproj]
> ```
> 
> Will submit a separate PR to hide those GCC pragmas behind '#ifdef **GNUC**'

See #837

**Comment by fhuberts** [comment]
I don't get it: it trips on the GCC pragmas, then I hide them and it trips on deprecated.

**Comment by fhuberts** [comment]
I've pulled out the commit "Wrap GCC pragma in '#ifdef __GNUC__'"

**Review by lganzzzo — CHANGES_REQUESTED** [review]
Windows build is failing with the following error:

```
D:\a\1\s\src\oatpp\core\data\mapping\type\Primitive.cpp(48,62): error C2061: syntax error: identifier 'ssize_t' [D:\a\1\s\build\src\oatpp.vcxproj]
D:\a\1\s\src\oatpp\core\data\mapping\type\Primitive.cpp(58,41): error C2061: syntax error: identifier 'ssize_t' [D:\a\1\s\build\src\oatpp.vcxproj]
```

it doesn't like `ssize_t`

File: src/oatpp/core/data/mapping/type/Primitive.cpp [hunk 1]
```diff
@@ -45,7 +45,7 @@ String String::loadFromFile(const char* filename) {
   if (file.is_open()) {
     auto result = data::mapping::type::String(file.tellg());
     file.seekg(0, std::ios::beg);
-    file.read(const_cast<char*>(result->data()), result->size());
+    file.read(const_cast<char*>(result->data()), static_cast<ssize_t>(result->size()));
```
[the remark this answers is not in the document]
Comment (fhuberts) [reply src/oatpp/core/data/mapping/type/Primitive.cpp:48]: will take a look

File: src/oatpp/core/data/mapping/type/Primitive.cpp [hunk 2]
```diff
@@ -55,7 +55,7 @@ String String::loadFromFile(const char* filename) {
 void String::saveToFile(const char* filename) const {
   std::ofstream fs(filename, std::ios::out | std::ios::binary);
   if(m_ptr != nullptr) {
-    fs.write(m_ptr->data(), m_ptr->size());
+    fs.write(m_ptr->data(), static_cast<ssize_t>(m_ptr->size()));
```
[the remark this answers is not in the document]
Comment (fhuberts) [reply src/oatpp/core/data/mapping/type/Primitive.cpp:58]: will take a look

**Comment by fhuberts** [comment]
sorry, removed the wrong branch from my fork. re-instated it, and will open new PR. sorry.

**Comment by fhuberts** [comment]
see #842

**Pull request closed without merging**